File: java_i/tp3/src/Exercise05.java
import market.Product;

public class Exercise05 implements IExercise {
  @Override
  public String getName() {
    return "Criando Métodos de Propriedade (Getters e Setters)";
  }

  @Override
  public void run() {
    var pepsiCan = new Product("Pepsi Black Lata", 2.77, 23);

    System.out.println("Nome do produto: " + pepsiCan.getName());
    System.out.println("Valor do produto: R$ " + pepsiCan.getPrice());

    pepsiCan.setPrice(2.50);
    System.out.println("Novo valor do produto: R$ " + pepsiCan.getPrice());
  }
}

File: c_sharp_i/TP3/Exercise05.cs
namespace TP3;

public class Exercise05 : IExercise
{
  public string Name => "Criando Métodos de Propriedade (Getters e Setters)";

  public void Run(string[] args)
  {
    var ticket = new Ticket
    {
      Price = 20f,
      Name = "Guns'n'Roses",
      Quantity = 100
    };

    var price = ticket.GetPrice();
    Console.WriteLine($"Preço do ingresso: {price}");

    ticket.SetPrice(39.99);

    price = ticket.GetPrice();
    Console.WriteLine($"Preço atualizado do ingresso: {price}");
  }
}
